### PR TITLE
Add table_aws_eventbridge_rule. Closes #113

### LINKED ISSUE #113: Add table_aws_eventbridge_rule

### PR DESCRIPTION
**Pull request by anisadas**

### CODE REVIEW

**Comment by anisadas** [comment]
```
SETUP: tests/aws_eventbridge_rule []

PRETEST: tests/aws_eventbridge_rule

TEST: tests/aws_eventbridge_rule
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
aws_cloudwatch_event_rule.named_test_resource: Creating...
aws_cloudwatch_event_rule.named_test_resource: Creation complete after 5s [id=turbottest973]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:events:us-east-1:986325076436:rule/turbottest973
resource_name = turbottest973

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:events:us-east-1:986325076436:rule/turbottest973",
    "event_pattern": "{\"detail-type\":[\"AWS Console Sign In via CloudTrail\"]}",
    "name": "turbottest973",
    "tags": {
      "name": "turbottest973"
    }
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "arn": "arn:aws:events:us-east-1:986325076436:rule/turbottest973",
    "name": "turbottest973",
    "tags_src": [
      {
        "Key": "name",
        "Value": "turbottest973"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:events:us-east-1:986325076436:rule/turbottest973",
    "event_bus_name": "default",
    "name": "turbottest973",
    "partition": "aws",
    "region": "us-east-1",
    "state": "ENABLED",
    "tags": {
      "name": "turbottest973"
    },
    "title": "turbottest973"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:events:us-east-1:986325076436:rule/turbottest973"
    ],
    "name": "turbottest973",
    "region": "us-east-1",
    "tags": {
      "name": "turbottest973"
    },
    "title": "turbottest973"
  }
]
✔ PASSED

POSTTEST: tests/aws_eventbridge_rule

TEARDOWN: tests/aws_eventbridge_rule

SUMMARY:

1/1 passed.
```

File: aws-test/tests/aws_eventbridge_rule/variables.tf [hunk 1]
```diff
@@ -0,0 +1,80 @@
+
+
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20200125-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+
+
```
Comment (Paulami30) [inline aws-test/tests/aws_eventbridge_rule/variables.tf:10]:
```suggestion
```

File: aws-test/tests/aws_eventbridge_rule/variables.tf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
```
Comment (Paulami30) [inline aws-test/tests/aws_eventbridge_rule/variables.tf:2]:
```suggestion
```

File: aws/table_aws_eventbridge_rule.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+			{
+				Name:        "description",
+				Description: "The description of the rule.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "name",
+				Description: "The name of the rule.",
+				Type:        proto.ColumnType_STRING,
+			},
```
Comment (Paulami30) [inline aws/table_aws_eventbridge_rule.go:40]: Make `name` as the first column

File: aws/table_aws_eventbridge_rule.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+			{
+				Name:        "tags_src",
+				Description: "A list of tags assigned to the rule.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsEventBridgeRuleTags,
+				Transform:   transform.FromField("Tags"),
+			},
+			/// Standard columns for all tables
```
Comment (Paulami30) [inline aws/table_aws_eventbridge_rule.go:80]:
```suggestion

			// Standard columns for all tables
```

File: docs/tables/aws_eventbridge_rule.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    arn,
+    state,
+    created_by,
+    event_bus_name
+from
+    aws_eventbridge_rule;
+```
+
+
+### List of rules which are not in enable state
```
Comment (Paulami30) [inline docs/tables/aws_eventbridge_rule.md:21]:
```suggestion
### List of rules which are not enabled
```

File: aws/table_aws_eventbridge_rule.go [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+	// Create Session
+	svc, err := EventBridgeService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &eventbridge.ListTagsForResourceInput{
+		//ResourceType: types.String("Parameter"),
```
Comment (Paulami30) [inline aws/table_aws_eventbridge_rule.go:205]:
```suggestion
	
```

File: aws/table_aws_eventbridge_rule.go [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+		}
+		if response.NextToken == nil {
+			break
+		}
+		param.NextToken = response.NextToken
+	}
+
+	return nil, err
+}
+
```
Comment (Paulami30) [inline aws/table_aws_eventbridge_rule.go:146]:
```suggestion

//// HYDRATE FUNCTIONS

```

File: docs/tables/aws_eventbridge_rule.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+select
+    name,
+    arn,
+    state,
+    created_by,
+    event_bus_name
+from
+    aws_eventbridge_rule;
+```
+
```
Comment (Paulami30) [inline docs/tables/aws_eventbridge_rule.md:19]:
```
select
  name,
  arn,
  state,
  created_by,
  event_bus_name
from
  aws_eventbridge_rule;
```

File: docs/tables/aws_eventbridge_rule.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  state,
+  created_by
+from
+  aws_eventbridge_rule
+where
+  state != 'ENABLED';
+```
+
+
+### List of targets info which are associated with rule
```
Comment (Paulami30) [inline docs/tables/aws_eventbridge_rule.md:36]:
```suggestion
### List of targets info associated with the rule
```

File: docs/tables/aws_eventbridge_rule.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+```sql
+select
+  name,
+  cd ->> 'Arn' as target_arn,
+  cd ->> 'RoleArn' as role_arn,
+  cd ->> 'Id' as id
```
Comment (Paulami30) [inline docs/tables/aws_eventbridge_rule.md:43]: which id is this? change the alias as a meaningful name.

File: docs/tables/aws_eventbridge_rule.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+from
+  aws_eventbridge_rule,
+  jsonb_array_elements(targets) as cd;
+```
+
+
+### List of rules which have iam role
```
Comment (Paulami30) [inline docs/tables/aws_eventbridge_rule.md:50]:
```suggestion
### List of rules which are not associated with any iam role
```

File: docs/tables/aws_eventbridge_rule.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+```sql
+select
+  name,
+  cd ->> 'RoleArn' as role_arn
+from
+  aws_eventbridge_rule,
+  jsonb_array_elements(targets) as cd
+where
+  cd ->> 'RoleArn' != 'null';
```
Comment (Paulami30) [inline docs/tables/aws_eventbridge_rule.md:60]:
```suggestion
  cd ->> 'RoleArn' = null;
```

**Review by Paulami30 — DISMISSED** [review]

File: aws/table_aws_eventbridge_rule.go [hunk 13]
```diff
@@ -0,0 +1,270 @@
...
+			},
+			{
+				Name:        "event_bus_name",
+				Description: "The name or ARN of the event bus associated with the rule.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "event_pattern",
+				Description: "The event pattern of the rule.",
+				Type:        proto.ColumnType_STRING,
```
Comment (Subhajit97) [inline aws/table_aws_eventbridge_rule.go:54]: Can we modify the `event_pattern` value from JSON string to JSON object?

File: aws/table_aws_eventbridge_rule.go [hunk 14]
```diff
@@ -0,0 +1,270 @@
...
+				Hydrate:     getAwsEventBridgeRule,
+			},
+			{
+				Name:        "targets",
```
Comment (Subhajit97) [inline aws/table_aws_eventbridge_rule.go:58]: Add all the JSON type column at the end

File: aws/table_aws_eventbridge_rule.go [hunk 15]
```diff
@@ -0,0 +1,270 @@
...
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsEventBridgeRuleTags")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	arn := *h.Item.(*eventbridge.DescribeRuleOutput).Arn
```
Comment (Subhajit97) [inline aws/table_aws_eventbridge_rule.go:231]:
```suggestion
	arn := h.Item.(*eventbridge.DescribeRuleOutput).Arn
```

File: aws/table_aws_eventbridge_rule.go [hunk 16]
```diff
@@ -0,0 +1,270 @@
...
+
+	// Create Session
+	svc, err := EventBridgeService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &eventbridge.ListTagsForResourceInput{
+		ResourceARN: &arn,
```
Comment (Subhajit97) [inline aws/table_aws_eventbridge_rule.go:241]:
```suggestion
		ResourceARN: arn,
```

File: aws/table_aws_eventbridge_rule.go [hunk 17]
```diff
@@ -0,0 +1,270 @@
...
+
+	return op, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func eventBridgeTagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("eventBridgeTagListToTurbotTags")
+	tagList := d.HydrateItem.(*eventbridge.ListTagsForResourceOutput)
+
```
Comment (Subhajit97) [inline aws/table_aws_eventbridge_rule.go:259]:
```suggestion

if tagList == nil {
  return nil, nil
}

```

File: aws/table_aws_eventbridge_rule.go [hunk 18]
```diff
@@ -0,0 +1,270 @@
...
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+func tableAwsEventBridge(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_eventbridge_rule",
+		Description: "AWS EventBridge Rule ",
```
Comment (Subhajit97) [inline aws/table_aws_eventbridge_rule.go:15]:
```suggestion
		Description: "AWS EventBridge Rule",
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
Hi @anisadas, I think we should update the `event_pattern` value from a JSON string to JSON object, which is easy to read.
Also there are some minor changes. Make the suggested changes and add the tests logs and the output of the queries added in docs.

**Comment by anisadas** [comment]
<details><summary>Results</summary>
<p>

```
select
  name,
  arn,
  state,
  created_by,
  event_bus_name
from
  aws_eventbridge_rule;

+------------------------------------------------+-------------------------------------------------------------------------------------------+-
| name                                           | arn                                                                                       | 
+------------------------------------------------+-------------------------------------------------------------------------------------------+-
| turbot_aws_api_events_kms                      | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_kms                      | 
| Amazon_Inspector_Assessment_0-4TegljLN_IddROk4 | arn:aws:events:us-east-2:986325076436:rule/Amazon_Inspector_Assessment_0-4TegljLN_IddROk4 | 
| turbot_aws_api_events_eks                      | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_eks                      | 
| Amazon_Inspector_Assessment_0-yiEZ2sUv_IdX6pZb | arn:aws:events:us-east-2:986325076436:rule/Amazon_Inspector_Assessment_0-yiEZ2sUv_IdX6pZb | 
| turbot_aws_api_events_cloudtrail               | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_cloudtrail               | 
| AutoScalingManagedRule                         | arn:aws:events:us-east-2:986325076436:rule/AutoScalingManagedRule                         | 
| turbot_aws_api_events_codecommit               | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_codecommit               | 
| turbot_aws_api_events_lambda                   | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_lambda                   | 
| turbot_aws_api_events_iam                      | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_iam                      | 
| turbot_aws_api_events_s3                       | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_s3                       | 
| turbot_aws_api_events_athena                   | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_athena                   | 
| turbot_aws_api_events_rds                      | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_rds                      | 
| turbot_aws_api_events_cloudformation           | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_cloudformation           | 
| turbot_aws_api_events_lightsail                | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_lightsail                | 
| turbot_aws_api_events_config                   | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_config                   | 
| turbot_aws_api_events_vpc_core                 | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_vpc_core                 | 
| turbot_aws_api_events_sns                      | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_sns                      | 
| turbot_aws_api_events_ecs                      | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_ecs                      | 
| turbot_aws_api_events_ecr                      | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_ecr                      | 
| turbot_aws_api_events_vpc_security             | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_vpc_security             | 
| turbot_aws_api_events_ssm                      | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_ssm                      | 
| turbot_aws_api_events_billingconsole           | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_billingconsole           | 
| turbot_aws_api_events_codebuild                | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_codebuild                | 
| turbot_aws_api_events                          | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events                          | 
| turbot_aws_api_events_s3                       | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_s3                       | 
| turbot_aws_api_events_logs                     | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_logs                     | 
| rupzz                                          | arn:aws:events:us-east-2:986325076436:rule/rupzz                                          | 
| turbot_aws_api_events_sns                      | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_sns                      | 
| turbot_aws_api_events_codecommit               | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_codecommit               | 
| turbot_aws_api_events_logs                     | arn:aws:events:us-east-1:986325076436:rule/turbot_aws_api_events_logs                     | 
| turbot_aws_api_events_iam                      | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_iam                      | 
| bug-fix                                        | arn:aws:events:us-east-2:986325076436:rule/bug-fix                                        | 
| turbot_aws_api_events_vpc_core                 | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_vpc_core                 | 
| turbot_aws_api_events_cloudtrail               | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_cloudtrail               | 
| turbot_aws_api_events_ssm                      | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_ssm                      | 
| turbot_aws_api_events_kms                      | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_kms                      | 
| turbot_aws_api_events_billingconsole           | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_billingconsole           | 
| turbot_aws_api_events_ecr                      | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_ecr                      | 
| turbot_aws_api_events_lambda                   | arn:aws:events:us-east-2:986325076436:rule/turbot_aws_api_events_lambda                   | 

---------+--------------+----------------+
 state   | created_by   | event_bus_name |
---------+--------------+----------------+
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |
 ENABLED | 986325076436 | default        |


select
  name,
  arn,
  state,
  created_by
from
  aws_eventbridge_rule
where
  state != 'ENABLED';

+----------+-----------------------------------------------------+----------+--------------+
| name     | arn                                                 | state    | created_by   |
+----------+-----------------------------------------------------+----------+--------------+
| test-stp | arn:aws:events:us-east-2:986325076436:rule/test-stp | DISABLED | 986325076436 |
+----------+-----------------------------------------------------+----------+--------------+

select
  name,
  cd ->> 'Id' as target_id,
  cd ->> 'Arn' as target_arn,
  cd ->> 'RoleArn' as role_arn
from
  aws_eventbridge_rule,
  jsonb_array_elements(targets) as cd;


+------------------------------------------------+----------------------------------------+----------------------------------------------------
| name                                           | target_id                              | target_arn                                         
+------------------------------------------------+----------------------------------------+----------------------------------------------------
| turbot_aws_api_events                          | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_billingconsole           | turbot_event_handlers                  | arn:aws:sns:us-east-2:986325076436:turbot_aws_api_h
| turbot_aws_api_events_ecs                      | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events                          | turbot_event_handlers                  | arn:aws:sns:us-east-2:986325076436:turbot_aws_api_h
| turbot_aws_api_events_vpc_security             | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| AutoScalingManagedRule                         | autoscaling                            | arn:aws:autoscaling:us-east-2:::                   
| turbot_aws_api_events_billingconsole           | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_sns                      | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_iam                      | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_codebuild                | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_ecr                      | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_cloudtrail               | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| bug-find                                       | Id776062935923                         | arn:aws:sns:us-east-1:986325076436:aws-console-logi
| bug-find                                       | Id9849873483892                        | arn:aws:lambda:us-east-1:986325076436:function:chec
| bug-find                                       | Id9986391734177                        | arn:aws:sns:us-east-1:986325076436:dynamodb        
| turbot_aws_api_events_codecommit               | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_athena                   | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_config                   | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_s3                       | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_cloudformation           | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_logs                     | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_eks                      | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_ssm                      | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_lightsail                | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_vpc_core                 | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_lambda                   | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_rds                      | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_kms                      | turbot_event_handlers                  | arn:aws:sns:us-east-1:986325076436:turbot_aws_api_h
| turbot_aws_api_events_eks                      | turbot_event_handlers                  | arn:aws:sns:us-east-2:986325076436:turbot_aws_api_h
| turbot_aws_api_events_athena                   | turbot_event_handlers                  | arn:aws:sns:us-east-2:986325076436:turbot_aws_api_h
| turbot_aws_api_events_s3                       | turbot_event_handlers                  | arn:aws:sns:us-east-2:986325076436:turbot_aws_api_h
| turbot_aws_api_events_cloudformation           | turbot_event_handlers                  | arn:aws:sns:us-east-2:986325076436:turbot_aws_api_h
| turbot_aws_api_events_codebuild                | turbot_event_handlers                  | arn:aws:sns:us-east-2:986325076436:turbot_aws_api_h
| Amazon_Inspector_Assessment_0-4TegljLN_IddROk4 | IddROk4                                | arn:aws:inspector:us-east-2:986325076436:target/0-L
| test-stp                                       | Idec9c5810-6e02-4403-83af-874b01ae19c6 | arn:aws:lambda:us-east-2:986325076436:function:api_
| turbot_aws_api_events_kms                      | turbot_event_handlers                  | arn:aws:sns:us-east-2:986325076436:turbot_aws_api_h
| bug-fix                                        | Id11855043398315                       | arn:aws:lambda:us-east-2:986325076436:function:Test
| turbot_aws_api_events_codecommit               | turbot_event_handlers                  | arn:aws:sns:us-east-2:986325076436:turbot_aws_api_h
| Amazon_Inspector_Assessment_0-yiEZ2sUv_IdX6pZb | IdX6pZb                                | arn:aws:inspector:us-east-2:986325076436:target/0-L

+-------------------------------------------------------------------------------+
                             | role_arn                                                                      |
-----------------------------+-------------------------------------------------------------------------------+
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
                             | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
ins-011                      | <null>                                                                        |
ck-sort                      | <null>                                                                        |
                             | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
handler                      | <null>                                                                        |
LUcbCx3h/template/0-4TegljLN | arn:aws:iam::986325076436:role/AWS_InspectorEvents_Invoke_Assessment_Template |
_gateway_authorizer          | <null>                                                                        |
handler                      | <null>                                                                        |
tLambda                      | <null>                                                                        |
handler                      | <null>                                                                        |
LVyai0k3/template/0-yiEZ2sUv | arn:aws:iam::986325076436:role/AWS_InspectorEvents_Invoke_Assessment_Template |

select
  name,
  cd ->> 'RoleArn' as role_arn
from
  aws_eventbridge_rule,
  jsonb_array_elements(targets) as cd
where
  cd ->> 'RoleArn' is null;

+--------------------------------------+----------+
| name                                 | role_arn |
+--------------------------------------+----------+
| turbot_aws_api_events_cloudformation | <null>   |
| turbot_aws_api_events                | <null>   |
| turbot_aws_api_events_vpc_security   | <null>   |
| turbot_aws_api_events_ecr            | <null>   |
| turbot_aws_api_events_codecommit     | <null>   |
| bug-find                             | <null>   |
| bug-find                             | <null>   |
| bug-find                             | <null>   |
| turbot_aws_api_events_s3             | <null>   |
| turbot_aws_api_events_lightsail      | <null>   |
| turbot_aws_api_events_eks            | <null>   |
| turbot_aws_api_events_cloudtrail     | <null>   |
| turbot_aws_api_events_kms            | <null>   |
| turbot_aws_api_events_logs           | <null>   |
| turbot_aws_api_events_sns            | <null>   |
| turbot_aws_api_events_iam            | <null>   |
| turbot_aws_api_events_codebuild      | <null>   |
| turbot_aws_api_events_rds            | <null>   |
| turbot_aws_api_events_lambda         | <null>   |
| turbot_aws_api_events_ecs            | <null>   |
| turbot_aws_api_events_athena         | <null>   |
| turbot_aws_api_events_billingconsole | <null>   |
| turbot_aws_api_events_config         | <null>   |
| turbot_aws_api_events_ssm            | <null>   |
| turbot_aws_api_events_ecr            | <null>   |
| AutoScalingManagedRule               | <null>   |
| turbot_aws_api_events_sns            | <null>   |
| turbot_aws_api_events_vpc_core       | <null>   |
| turbot_aws_api_events_iam            | <null>   |
| turbot_aws_api_events_vpc_security   | <null>   |
| turbot_aws_api_events_eks            | <null>   |
| turbot_aws_api_events_logs           | <null>   |
| turbot_aws_api_events_cloudtrail     | <null>   |
| turbot_aws_api_events_athena         | <null>   |
| bug-fix                              | <null>   |
| turbot_aws_api_events_billingconsole | <null>   |
| turbot_aws_api_events_cloudformation | <null>   |
| test-stp                             | <null>   |
```

</p>
</details>

**Review by Subhajit97 — APPROVED** [review]
LGTM

File: aws/plugin.go [hunk 19]
```diff
@@ -63,6 +63,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_ec2_transit_gateway":                tableAwsEc2TransitGateway(ctx),
 			"aws_ec2_transit_gateway_route_table":    tableAwsEc2TransitGatewayRouteTable(ctx),
 			"aws_ec2_transit_gateway_vpc_attachment": tableAwsEc2TransitGatewayVpcAttachment(ctx),
+			"aws_eventbridge_rule":                   tableAwsEventBridge(ctx),
```
Comment (cbruno10) [inline aws/plugin.go:66]:
```suggestion
			"aws_eventbridge_rule":                   tableAwsEventBridgeRule(ctx),
```

File: aws/table_aws_eventbridge_rule.go [hunk 20]
```diff
@@ -0,0 +1,274 @@
...
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+func tableAwsEventBridge(_ context.Context) *plugin.Table {
```
Comment (cbruno10) [inline aws/table_aws_eventbridge_rule.go:12]:
```suggestion
func tableAwsEventBridgeRule(_ context.Context) *plugin.Table {
```

File: docs/tables/aws_eventbridge_rule.md [hunk 21]
```diff
@@ -0,0 +1,62 @@
+# Table: aws_eventbridge_rule
+
+Amazon EventBridge defines the building and management of event-driven applications by taking care of event ingestion and delivery, security, authorization, and error-handling.
+
+## Examples
+
+### EventBridge rule basic info
```
Comment (cbruno10) [inline docs/tables/aws_eventbridge_rule.md:7]:
```suggestion
### Basic info
```

File: docs/tables/aws_eventbridge_rule.md [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+  arn,
+  state,
+  created_by,
+  event_bus_name
+from
+  aws_eventbridge_rule;
+```
+
+
+### List of rules which are not enabled
```
Comment (cbruno10) [inline docs/tables/aws_eventbridge_rule.md:21]:
```suggestion
### List disabled rules
```

File: docs/tables/aws_eventbridge_rule.md [hunk 23]
```diff
@@ -0,0 +1,62 @@
...
+  state,
+  created_by
+from
+  aws_eventbridge_rule
+where
+  state != 'ENABLED';
+```
+
+
+### List of targets and IAM roles associated with the rule
```
Comment (cbruno10) [inline docs/tables/aws_eventbridge_rule.md:36]:
```suggestion
### Get the targets and IAM role associated with each rule
```

File: docs/tables/aws_eventbridge_rule.md [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+  cd ->> 'Id' as target_id,
+  cd ->> 'Arn' as target_arn,
+  cd ->> 'RoleArn' as role_arn
+from
+  aws_eventbridge_rule,
+  jsonb_array_elements(targets) as cd;
+```
+
+
+### List of rules which are not associated with any iam role
```
Comment (cbruno10) [inline docs/tables/aws_eventbridge_rule.md:50]: What's the impact of a rule not being associated with an IAM role? Does the rule not work then?

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**